abrt/faf
# Semantic versioning

storage: add Build.semver (semantic version field)
    
    Requirement for comparing versions on database level.
    
    Uses pg-semver (https://github.com/theory/pg-semver/)
    PostgreSQL extension implementing Semantic version
    specification (http://semver.org/).
    
    Closes #334.
 
Also contains commits from PR #388 as it depends on it (these commits will disappear as it's merged).

How to test:

1) install packages

2) as admin user of postgresql run 

    CREATE EXTENSION semver;

this can't be done via migration as only admin is permitted to install extensions.

3) run migration
4) verify via 

    SELECT * FROM builds ORDER by base_package_name, semver DESC LIMIT 1000;

or via `faf shell`

    session.query(Build.semver).order_by(Build.semver).limit(100).all()
